### PR TITLE
feat: update system.asserts to use new Assert class - Data Recipes

### PR DESCRIPTION
**Pull request by msrivastav13**
### What does this PR do?

Update asserts to use new winter23 Assert class. The new Assert class is really elegant compared to the older System methods.

### What issues does this PR fix or reference?

#<Insert GitHub Issue>

## The PR fulfills these requirements:

[ ] Tests for the proposed changes have been added/updated.
[ ] Code linting and formatting was performed.

### Functionality Before

<insert gif and/or summary>

### Functionality After

<insert gif and/or summary>

### CODE REVIEW

File: force-app/tests/Data Recipes/DMLRecipes_Tests.cls [hunk 1]
```diff
@@ -57,11 +57,11 @@ public class DMLRecipes_Tests {
             WHERE name = 'Texas'
         ];
 
-        System.assert(
+        System.Assert.isTrue(
```
Comment (pozil) [inline force-app/tests/Data Recipes/DMLRecipes_Tests.cls:60]: Change to `areEqual`

File: force-app/tests/Data Recipes/DMLRecipes_Tests.cls [hunk 2]
```diff
@@ -57,11 +57,11 @@ public class DMLRecipes_Tests {
...
             acct.size() == 1,
             'Expected the list of accounts to have a single account'
         );
-        System.assert(
+        System.Assert.isTrue(
```
Comment (pozil) [inline force-app/tests/Data Recipes/DMLRecipes_Tests.cls:64]: Change to `areEqual`

File: force-app/tests/Data Recipes/DMLRecipes_Tests.cls [hunk 3]
```diff
@@ -357,7 +357,7 @@ public class DMLRecipes_Tests {
             FROM Account
             WHERE id = :acct.Id
         ];
-        System.assert(
+        System.Assert.isTrue(
```
Comment (pozil) [inline force-app/tests/Data Recipes/DMLRecipes_Tests.cls:360]: Change to `areEqual`

File: force-app/tests/Data Recipes/DMLRecipes_Tests.cls [hunk 4]
```diff
@@ -396,7 +396,7 @@ public class DMLRecipes_Tests {
...
```
Comment (pozil) [inline force-app/tests/Data Recipes/DMLRecipes_Tests.cls:399]: Change to `areEqual`

File: force-app/tests/Data Recipes/DynamicSOQLRecipes_Tests.cls [hunk 5]
```diff
@@ -13,7 +13,10 @@ private inherited sharing class DynamicSOQLRecipes_Tests {
         List<Account> results = DynamicSOQLRecipes.simpleDynamicSOQLQuery();
         Test.stopTest();
 
-        System.assert(results.size() == 1, 'Expected to find a single account');
+        System.Assert.isTrue(
```
Comment (pozil) [inline force-app/tests/Data Recipes/DynamicSOQLRecipes_Tests.cls:16]: Change to `areEqual`

Comment (pozil) [reply force-app/tests/Data Recipes/DynamicSOQLRecipes_Tests.cls:16]: Same remark applies to the other tests below.

File: force-app/tests/Data Recipes/SOQLRecipes_Tests.cls [hunk 6]
```diff
@@ -62,20 +62,20 @@ private class SOQLRecipes_Tests {
     static void testSimplegetRecordsPositive() {
         TestDataHelpers.createAccount(false, '');
 
-        System.assertEquals(
+        System.Assert.areEqual(
```
Comment (pozil) [inline force-app/tests/Data Recipes/SOQLRecipes_Tests.cls:65]: Fix param order

Comment (pozil) [reply force-app/tests/Data Recipes/SOQLRecipes_Tests.cls:65]: Same remark applies to the several tests below.

**Review by pozil — CHANGES_REQUESTED** [review]
Found several issues.

File: force-app/tests/Data Recipes/DynamicSOQLRecipes_Tests.cls (hunk 5 above)
Comment (msrivastav13) [reply force-app/tests/Data Recipes/DynamicSOQLRecipes_Tests.cls]: Fixed!

File: force-app/tests/Data Recipes/DMLRecipes_Tests.cls (hunk 4 above)
Comment (msrivastav13) [reply force-app/tests/Data Recipes/DMLRecipes_Tests.cls]: Fixed!

**Review by pozil — APPROVED** [review]
All good, thanks.

**Pull request merged**